### PR TITLE
[AAE-1948] Form Attach File cloud widget E2E

### PR DESCRIPTION
**Pull request by sivakumar414ram**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

https://issues.alfresco.com/jira/browse/AAE-1948

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+        await expect(await processCloudDemoPage.processFilterCloudComponent.getActiveFilterName()).toBe('Completed Processes');
+        await processCloudDemoPage.processListCloudComponent().checkContentIsDisplayedByName(processName);
+    });
+
+    async function attachFileFromContent(fileName: string) {
+        const contentUploadFileWidget = await processCloudWidget.attachFileWidgetCloud(widgets.contentMultipleAttachFileId);
+        await contentUploadFileWidget.clickAttachContentFile(widgets.contentMultipleAttachFileId);
+
+        await contentNodeSelectorDialog.checkDialogIsDisplayed();
```
Comment (cristinaj) [inline e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts]: Thinking that this method could be useful for some other tests, i would recommend to move it[lines 127-144] under contentNodeSelectorDialog PO.

File: e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    const contentNodeSelectorDialog = new ContentNodeSelectorDialogPage();
+    const viewerPage = new ViewerPage();
+
+    const processName = StringUtil.generateRandomString(10);
+    const simpleApp = browser.params.resources.ACTIVITI_CLOUD_APPS.SIMPLE_APP.name;
+    const folderName = StringUtil.generateRandomString(5);
+    let uploadedFolder: any;
+
+    const widgets = {
+        contentMultipleAttachFileId: 'UploadMultipleFileFromContentId'
```
Comment (cristinaj) [inline e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts]: Can this value be saved into resources file?

Comment (sivakumar414ram) [reply e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts]: Updated

File: e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts (hunk 1 above)
Comment (sivakumar414ram) [reply e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts]: Added as part of contentNodeSelectorDialog page. Thanks

File: lib/testing/src/lib/content-services/dialog/content-node-selector-dialog.page.ts [hunk 3]
```diff
@@ -108,6 +108,27 @@ export class ContentNodeSelectorDialogPage {
         await this.contentList.dataTablePage().doubleClickRowByContent(name);
     }
 
+    async attachFileFromContentNode(folderName: string, fileName: string) {
+        await this.checkDialogIsDisplayed();
+        await this.checkSearchInputIsDisplayed();
+        await this.checkCancelButtonIsDisplayed();
+
+        const contentList = await this.contentListPage();
```
Comment (cristinaj) [inline lib/testing/src/lib/content-services/dialog/content-node-selector-dialog.page.ts]: Can contentList declared at line 33 be used here too?

File: e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        await startProcessPage.startProcessWithProcessDefinition(processName, processDefinitionName);
+
+        await processCloudDemoPage.processFilterCloudComponent.clickOnProcessFilters();
+        await processCloudDemoPage.processFilterCloudComponent.clickRunningProcessesFilter();
+        await expect(await processCloudDemoPage.processFilterCloudComponent.getActiveFilterName()).toBe('Running Processes');
+        await processCloudDemoPage.processListCloudComponent().checkContentIsDisplayedByName(processName);
+        await processCloudDemoPage.processListCloudComponent().selectRow(processName);
+
+        await tasksCloudDemoPage.taskListCloudComponent().checkTaskListIsLoaded();
+        await tasksCloudDemoPage.taskListCloudComponent().selectRow(taskName);
```
Comment (cristinaj) [inline e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts]: I would suggest to create the process instance using API call and navigate directly to the task created.

File: lib/testing/src/lib/content-services/dialog/content-node-selector-dialog.page.ts (hunk 3 above)
Comment (adinapitul) [inline lib/testing/src/lib/content-services/dialog/content-node-selector-dialog.page.ts]: await needs to be removed, contentListPage() does not return a promise

Comment (sivakumar414ram) [reply lib/testing/src/lib/content-services/dialog/content-node-selector-dialog.page.ts]: Removed

Comment (sivakumar414ram) [reply lib/testing/src/lib/content-services/dialog/content-node-selector-dialog.page.ts]: Updated

File: e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts (hunk 4 above)
Comment (mauriziovitale) [reply e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts]: it makes sense. @sivakumar414ram

Comment (sivakumar414ram) [reply e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts]: Thanks.  i will update it

Comment (sivakumar414ram) [reply e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts]: Updated e2e (now process instance is getting created using API).

File: e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    ContentNodeSelectorDialogPage,
+    ProcessDefinitionsService,
+    ProcessInstancesService,
+    ApiService
+} from '@alfresco/adf-testing';
+import { ProcessCloudDemoPage } from '../pages/adf/demo-shell/process-services/process-cloud-demo.page';
+import { NavigationBarPage } from '../pages/adf/navigation-bar.page';
+import { TasksCloudDemoPage } from '../pages/adf/demo-shell/process-services/tasks-cloud-demo.page';
+
+describe('Process Task - Attache content file', () => {
```
Comment (adinapitul) [inline e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts]: misspelling here attache -> attach

File: e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+        await tasksCloudDemoPage.taskListCloudComponent().checkContentIsDisplayedByName(taskName);
+
+        await processCloudDemoPage.processFilterCloudComponent.clickOnProcessFilters();
+        await processCloudDemoPage.processFilterCloudComponent.clickCompletedProcessesFilter();
+
+        await expect(await processCloudDemoPage.processFilterCloudComponent.getActiveFilterName()).toBe('Completed Processes');
+        await processCloudDemoPage.processListCloudComponent().checkContentIsDisplayedById(processInstance.entry.id);
+    });
+
+    async function viewAttachedFile(contentUploadWidget, fileName: string) {
```
Comment (adinapitul) [inline e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts]: add return type

File: lib/testing/src/lib/content-services/dialog/content-node-selector-dialog.page.ts [hunk 7]
```diff
@@ -108,6 +108,25 @@ export class ContentNodeSelectorDialogPage {
         await this.contentList.dataTablePage().doubleClickRowByContent(name);
     }
 
+    async attachFileFromContentNode(folderName: string, fileName: string) {
```
Comment (adinapitul) [inline lib/testing/src/lib/content-services/dialog/content-node-selector-dialog.page.ts]: missing return type.

File: lib/testing/src/lib/content-services/dialog/content-node-selector-dialog.page.ts [hunk 8]
```diff
@@ -108,6 +108,25 @@ export class ContentNodeSelectorDialogPage {
...
+        await this.checkDialogIsDisplayed();
+        await this.checkSearchInputIsDisplayed();
+        await this.checkCancelButtonIsDisplayed();
+
+        await this.contentList.dataTablePage().waitForTableBody();
```
Comment (adinapitul) [inline lib/testing/src/lib/content-services/dialog/content-node-selector-dialog.page.ts]: you could define something like
`const dataTable = this.contentList.dataTablePage()`
and then use that variable on lines 117 -> 123

File: e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts (hunk 6 above)
Comment (sivakumar414ram) [reply e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts]: Added return type :+1:

File: lib/testing/src/lib/content-services/dialog/content-node-selector-dialog.page.ts (hunk 8 above)
Comment (sivakumar414ram) [reply lib/testing/src/lib/content-services/dialog/content-node-selector-dialog.page.ts]: updated

File: lib/testing/src/lib/content-services/dialog/content-node-selector-dialog.page.ts (hunk 7 above)
Comment (sivakumar414ram) [reply lib/testing/src/lib/content-services/dialog/content-node-selector-dialog.page.ts]: Updated

File: e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+    async function viewAttachedFile(contentUploadWidget, fileName: string): Promise<void> {
+        await contentUploadWidget.checkFileIsAttached(fileName);
+        await contentUploadWidget.viewFile(fileName);
+
+        await viewerPage.checkFileThumbnailIsDisplayed();
+        await viewerPage.checkFileNameIsDisplayed(fileName);
+        await viewerPage.clickCloseButton();
+    }
+
+    afterAll(async () => {
```
Comment (eromano) [inline e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts]: afterAll should be at the beginning

Comment (sivakumar414ram) [reply e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts]: Moved

File: e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+        'name': browser.params.resources.Files.ADF_DOCUMENTS.PDF_B.file_name,
+        'location': browser.params.resources.Files.ADF_DOCUMENTS.PDF_B.file_location
+    };
+
+    const apiService = new ApiService(browser.params.config.oauth2.clientId, browser.params.config.bpmHost, browser.params.config.oauth2.host, 'BPM');
+    this.alfrescoJsApi = new AlfrescoApi({ provider: 'ECM', hostEcm: browser.params.config.bpmHost });
+    const uploadActions = new UploadActions(this.alfrescoJsApi);
+
+    beforeAll(async () => {
+
```
Comment (eromano) [inline e2e/process-services-cloud/process-task-attach-content-file-cloud.e2e.ts]: not necessary space

File: lib/testing/src/lib/content-services/dialog/content-node-selector-dialog.page.ts [hunk 11]
```diff
@@ -99,13 +101,32 @@ export class ContentNodeSelectorDialogPage {
...
     async clickContentNodeSelectorResult(name): Promise<void> {
-        await this.contentList.dataTablePage().clickRowByContent(name);
+        await this.dataTable.clickRowByContent(name);
     }
 
     async doubleClickContentNodeSelectorResult(name): Promise<void> {
         // First click to select from search mode and second click to actually open node
-        await this.contentList.dataTablePage().doubleClickRowByContent(name);
-        await this.contentList.dataTablePage().doubleClickRowByContent(name);
+        await this.dataTable.doubleClickRowByContent(name);
```
Comment (adinapitul) [inline lib/testing/src/lib/content-services/dialog/content-node-selector-dialog.page.ts]: you're double-clicking twice here ?

Comment (sivakumar414ram) [reply lib/testing/src/lib/content-services/dialog/content-node-selector-dialog.page.ts]: this code was there already and I only changed this.contentList.dataTablePage() => this.dataTable. Btn it has a valid comment ( see line 108)

Comment (adinapitul) [reply lib/testing/src/lib/content-services/dialog/content-node-selector-dialog.page.ts]: I can see that the code was already there, but do you think it's correct to doubleClick twice?

Comment (mauriziovitale) [reply lib/testing/src/lib/content-services/dialog/content-node-selector-dialog.page.ts:109]: @adinapitul can you raise a different bug with more context? In this way, we can discuss if the new behaviour is correct

**Pull request merged**